### PR TITLE
Update libafl-based fuzzers (from AFL++ fork)

### PR DESCRIPTION
**Pull request by addisoncrump**
Changing forks so @tokatoka can collab with me on this. Supercedes #2021.

### CODE REVIEW

**Comment by addisoncrump** [comment]
@DonggeLiu When possible, could you please run a short experiment to see if the CI is spuriously failing? I think this will still be affected by #2023.

**Comment by tokatoka** [comment]
are you sure 799c634fef047d3e98355fe1ad17c5226c901a57 is the right commit?

**Comment by addisoncrump** [comment]
That was the state of `main` when I checked it... I also tested this locally.

**Comment by tokatoka** [comment]
and it looks like the problem is from `libafl_text` we should drop it

**Comment by addisoncrump** [comment]
Didn't know if we still cared about `libafl_text` :sweat_smile: I can delete.

**Comment by DonggeLiu** [comment]
/gcbrun run_experiment.py -a --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name 2024-08-15-2027-1 --fuzzers libafl libafl_libfuzzer

**Comment by DonggeLiu** [comment]
Experiment `2024-08-15-2027-1` data and results will be available later at:
[The experiment data](https://storage.googleapis.com/fuzzbench-data/index.html?prefix=2024-08-15-2027-1/).
[The experiment report](https://storage.googleapis.com/www.fuzzbench.com/reports/2024-08-15-2027-1/index.html).
[The experiment report(experimental)](https://storage.googleapis.com/www.fuzzbench.com/reports/experimental/2024-08-15-2027-1/index.html).

**Comment by tokatoka** [comment]
idk we only got data for two of the fuzzers?

**Comment by tokatoka** [comment]
wait but they do have the result for the 30 minutes run in the experiment-data, so everything is working i guess

File: benchmarks/harfbuzz_hb-shape-fuzzer_17863b/build.sh [hunk 1]
```diff
@@ -25,8 +25,8 @@ python3.8 -m pip install ninja meson==0.56.0
 
 # Disable:
 # 1. UBSan vptr since target built with -fno-rtti.
-export CFLAGS="$CFLAGS -fno-sanitize=vptr -DHB_NO_VISIBILITY"
-export CXXFLAGS="$CXXFLAGS -fno-sanitize=vptr -DHB_NO_VISIBILITY"
+export CFLAGS="$CFLAGS -fno-sanitize=vptr -DHB_NO_VISIBILITY -DHB_NO_PRAGMA_GCC_DIAGNOSTIC -Wno-cast-function-type-strict -Wno-incompatible-function-pointer-types-strict"
+export CXXFLAGS="$CXXFLAGS -fno-sanitize=vptr -DHB_NO_VISIBILITY -DHB_NO_PRAGMA_GCC_DIAGNOSTIC -Wno-cast-function-type-strict -Wno-incompatible-function-pointer-types-strict"
```
Comment (addisoncrump) [inline benchmarks/harfbuzz_hb-shape-fuzzer_17863b/build.sh:29]: This should now work for all clang-based fuzzers.

**Comment by tokatoka** [comment]
so it's ready to merge?

**Comment by addisoncrump** [comment]
Somehow the php bug benchmark has bitrotted during this PR!

I'll revert the experiment-relevant changes and then yes, it'll be good to go.

**Comment by DonggeLiu** [comment]
Strangely `FuzzBench` reports `libafl` failed to build with some benchmarks, but [the build log](file:///Users/donggeliu/Downloads/2024-08-15-2027-1_build-logs_benchmark-zlib_zlib_uncompress_fuzzer-fuzzer-libafl.txt) says otherwise:
image
But I don't think this relates to this PR, so let's merge.

Similarly, there are some coverage run failures, but it should not block this PR either:
image

**Pull request merged**